packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,170 @@
+
+/**
+ * Our policy is to support N/N-1 compatibility by default, where N is the most recent public major release of the runtime.
+ * Therefore, if the customer does not provide a compatibility mode, we will default to use N-1.
+ *
+ * However, this is not consistent with today's behavior. Some options (i.e. batching, compression) are enabled by default
+ * despite not being compatible with 1.x clients. Since the policy was introduced during 2.x's lifespan, N/N-1 compatibility
+ * by **default** will be in effect starting with 3.0. Importantly though, N/N-1 compatibility is still guaranteed with the
+ * proper configurations set.
+ *
+ * TODO: This should be updated to "2.0.0" when 3.0 is released.
+ */
+export const defaultCompatibilityMode = "pre-3.0-default";
+
+/**
+ * The current default set of configurations if no compatibility mode is provided.
+ * Since both compatibilityMode and the cross-client compat policy was introduced during 2.x's lifespan, we will maintain the
+ * set of default configurations that were in place before compatibilityMode and the policy were introduced.
+ * TODO: This can be removed after 3.0 is released.
+ */
+const defaultConfigsForPreFF3: IContainerRuntimeOptionsVersionDependent = {
+	gcOptions: {},
+	flushMode: FlushMode.TurnBased,
+	compressionOptions: enabledCompressionConfig,
+	enableRuntimeIdCompressor: undefined as unknown as "on" | "delayed",
+	enableGroupedBatching: true,
+	explicitSchemaControl: false,
+};
+
+/**
+ * Subset of the IContainerRuntimeOptionsInternal properties which are version-dependent.
+ *
+ * @remarks
+ * When a new option is added to IContainerRuntimeOptionsInternal, we must consider if it's a version-dependent option.
+ * If it's considered version-dependent, then a corresponding entry must be added to `versionDependentOptionConfigs`. If not, then
+ * it must be omitted from this type.
+ *
+ * Note: We use `Omit` instead of `Pick` to ensure that all new options are included in this type by default. If any new properties
+ * are added to IContainerRuntimeOptionsInternal, they will be included in this type unless explicitly omitted. This will prevent
+ * us from forgetting to account for any new properties in the future.
+ */
+export type IContainerRuntimeOptionsVersionDependent = Required<
+	Omit<
+		IContainerRuntimeOptionsInternal,
+		| "chunkSizeInBytes"
+		| "maxBatchSizeInBytes"
+		| "loadSequenceNumberVerification"
+		| "summaryOptions"
+		| "compatibilityMode"
+	>
+>;
+
+/**
+ * Mapping of version-dependent options to their compatibility related configs.
+ *
+ * Each key in this map corresponds to a property in IContainerRuntimeOptionsVersionDependent.
+ * The value is an object that maps a version string to the default value for that property when using compatibilityMode of at least that version.
+ */
+const versionDependentOptionConfigMap: {
+	[K in keyof IContainerRuntimeOptionsVersionDependent]: {
+		[version: string]: IContainerRuntimeOptionsVersionDependent[K];
```

Comment:
nit; we can have slightly better typing for version fields overall:

type SemanticVersion = `${number}.${number}.${number}` |  `${number}.${number}.${number}-{$string}`